rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,501 @@
+
+# Detailed design
+[design]: #detailed-design
+
+Each subsection here contains a decision along with arguments and counter-arguments for (+) and against (-) that decision.
+
+> Note: This RFC does not promote any tool; It focuses only on the format, not implementation. At the same time, this RFC must be technically feasible.
+
+## Discontinue nixdoc's custom format
+
+**Observing**: Nixdoc currently provides some format that can generate documentation from a small fraction of files in nixpkgs, Namely the files for `.lib` functions. The folder provides a consistent community convention that allows the building of static documentation. Applying nixdoc to files outside that directory is only partially possible due to the lack of a standard doc-comment format. 
+
+
+**Considering**: Should we propose the `nixdoc` format as a community standard?
+
+**Decision**: Discontinue nixdoc's custom format in favor of using something more generic.
+
+<details>
+<summary>Arguments</summary>
+
+- (+) Stays primarily compatible with the currently used comments in lib.
+    - (-) Outside the ./lib folder, nixodc is not a convention. 
+- (-) Is not widely understood.
+- (-) Is not widely used. (e.g., Outside the /lib folder)
+- (-) The format is specific, should be more generic, and follow established conventions. (e.g., Markdown, Doxygen, JSdoc, Haddock)
+- (-) The format is too complex and needs a whole list of complex rules to describe the exact format and content rules; this can also be observed in many false-formatted comments that do not render correctly.
+    - (-) The sections are implicitly continued.
+    - (-) The predefined section headers/keywords are very close to natural language and may accidentally be used.
+        - (-) Syntax to start new sections does not follow conventions (e.g., `Example:`) In contrast JSdoc defines `@Example`; Markdown `# Example`; Doxygen `@example`; Haskell-Haddock `>>>`
+    - (-) The content is implicitly rendered differently depending on predefined rules. Those rules need to be understood/feel natural.
+- (+) It Is already quite established and has some adoptions outside the `/lib` folder.
+- (-) Some multiline comments become doc-comments, but it is unclear which comments have such a property.
+    
+</details>
+
+## `/**` to start a doc-comment
+
+**Observing**: Doc-comments' outer format should be a distinctive subset of regular comments. Nevertheless, it should allow native writing without an ide or full-blown-language servers.
+
+**Considering**: `/** {content} */` where `{content}` is the inner format which is discussed later.
+
+**Decision**: use `/** {content} */` as the outer format.
+
+<details>
+<summary>Arguments</summary>
+
+- (+) It is mostly compatible with the currently used multiline comments.
+- (+) Is a strict subset of multiline comments, which allows multiline documentation natively.
+- (+) Does not need Language Server (LSP) support for productive usage. (In contrast to, e.g., `##`)
```

Comment:
Editor support, not LSP.

Also, this is the first mention of `##`, which means fresh readers of this RFC probably don't understand what this is about.